### ISSUE TITLE
Possible to specify directory for push / pull ?

### ISSUE THREAD

**Issue by jondcallahan** [google/clasp#22]
## Expected Behavior
I have my source files in an `src` directory, but do some transpiling which outputs the files to a `build` directory. Is it possible to specify a directory to push files from?

## Actual Behavior
Files are pushed and pulled from the root of the repo

## Specifications

  - Node version (`node -v`): any
  - Version (`npm list | grep clasp`): 1.0.7
  - OS (Mac/Linux/Windows): Mac

**Comment by grant** [comment]
Your `.clasp.json` file should be in the `build/` directory if you want that directory to be the root. I think this is an interesting feature but probably not one that I really see being useful.

Perhaps the README can describe how this works a bit more (for users that have different folders).

**Comment by jondcallahan** [comment]
Moving the `.clasp.json` file to the build directory makes it necessary to run the command from that directory. Would you be open to an optional arg passed to the push command specifying a directory? Something like:
```javascript
program
  .command('push [dir]')
  .description('Update the remote project')
  .action((dir = './') => { // use './' as a default
...
   let nameWithoutExt = name.slice(0, -path.extname(name).length);
   let formattedName = nameWithoutExt;
    if (dir !== './') {
      // Transforms build/appsscript to appsscript
      formattedName = formattedName.slice(dir.length + 1, formattedName.length);
    }
...
    return {
      name: formattedName, // the API separates the extension
```

**Comment by jondcallahan** [comment]
Another workaround for this use case is to ignore everything but the build directory in the `.claspignore`. This requires the manifest file be in the project root though which doesn't work if you have environment specific manifest files that get placed in the build dir.

**Comment by grant** [comment]
This is an interesting case. I'd be hesitant to add an option for `push`, in case one day you `clasp push` instead of `clasp push build`. Also, `clasp pull` would pull to the wrong dir.

It seems like the root directory should be a property of `.clasp.json` that defaults to `.` (same dir as the `.clasp.json` file). You normally won't push different directories all the time, so it should be a config, not a push option. Pulling should also respect the rootDir.

So `.clasp.json` could have:
```
{
  "scriptId": "1w0YR2N3Y9x_T3hbD_FGIrK5qEPATzFznF8IoNVk3enkVpkfzuU9AG4vE",
  "rootDir": "build"
}
```

If you want to work on this feature, I'll accept PRs.

**Comment by umihico** [comment]
I encountered the same issue and I'd like to share my personal experience with it.

I was using @google/aside and decided to switch projects. To do this, I removed the `.clasp.json` file and executed `clasp create [NEW_PROJECT_NAME]`. After this, the `rootDir` property of .clasp.json was set as project root, but the original configuration for aside points to `/.dist`. As a result, aside attempted to upload the entire project directory, including the node_modules.
